### PR TITLE
[SofaBase] Convert tests to Sofa.Testing

### PR DESCRIPTION
**Pull request by fredroy**
#1885 

+ some dispatch to SofaMeshCollision_test and SofaMiscCollision_test

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/BroadPhase_test.h [hunk 1]
```diff
@@ -84,7 +83,17 @@ struct MyBox{
...
+struct BroadPhaseTest: public ::testing::Test
+{
+    void SetUp() override
+    {
+        // force load SofaMiscCollision
+        sofa::component::initSofaMiscCollision();
+    }
+    void TearDown() override
+    {
+    }
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/BroadPhase_test.h:95]: empty you are sure? no need to clean graph?

Comment (epernod) [reply SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/BroadPhase_test.h:95]: if (root)
            Simulation::theSimulation->unload(root);

File: SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/SpherePrimitiveCreator.h [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+typedef sofa::component::container::MechanicalObject<sofa::defaulttype::Vec3Types> MechanicalObject3;
+typedef sofa::component::container::MechanicalObject<sofa::defaulttype::Rigid3Types> MechanicalObjectRigid3;
+
+using sofa::core::objectmodel::Data;
+using sofa::core::objectmodel::New;
+using sofa::component::container::MechanicalObject;
+using namespace sofa::defaulttype;
+
+namespace sofa
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/SpherePrimitiveCreator.h]: namespace sofa?  maybe a collision_test

File: SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/Sphere_test.cpp [hunk 3]
```diff
@@ -126,8 +118,8 @@ bool TestSphere::rigidRigid1(){
 
     sofa::helper::vector<DetectionOutput> detectionOUTPUT;
 
-    //loooking for an intersection
-    if(!BaseIntTool::computeIntersection(sph1,sph2,1.0,1.0,&detectionOUTPUT))
+    //looking for an intersection (with proximities)
+    if(!m_proxIntersection->computeIntersection(sph1,sph2,&detectionOUTPUT))
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/Sphere_test.cpp:122]: no check if m_proxIntersection is not null

File: modules/SofaMiscCollision/SofaMiscCollision_test/CMakeLists.txt [hunk 4]
```diff
@@ -4,13 +4,23 @@ project(SofaMiscCollision_test)
...
-add_executable(${PROJECT_NAME} ${SOURCE_FILES})
+add_executable(${PROJECT_NAME} ${HEADER_FILES} ${SOURCE_FILES})
 add_definitions("-DSOFAMISCCOLLISION_TEST_SCENES_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}/scenes\"")
 target_link_libraries(${PROJECT_NAME} PUBLIC Sofa.Testing SofaMiscCollision
         SofaBase SofaGeneralDeformable SofaImplicitOdeSolver SofaMeshCollision SofaTopologyMapping SofaLoader)
 
+get_target_property(SOFAMESHCOLLISIONTEST_INCLUDES SofaMeshCollision_test INCLUDE_DIRECTORIES)
+
+target_include_directories(${PROJECT_NAME} PUBLIC ${SOFAMESHCOLLISIONTEST_INCLUDES})
+target_include_directories(${PROJECT_NAME} PUBLIC ..)
```
Comment (epernod) [inline modules/SofaMiscCollision/SofaMiscCollision_test/CMakeLists.txt:24]: on purpose?

File: SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/BroadPhase_test.h (hunk 1 above)
Comment (fredroy) [reply SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/BroadPhase_test.h:95]: there is no created simulation actually in those tests.

File: SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/SpherePrimitiveCreator.h (hunk 2 above)
Comment (fredroy) [reply SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/SpherePrimitiveCreator.h]: yes why not

File: SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/Sphere_test.cpp (hunk 3 above)
Comment (fredroy) [reply SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/Sphere_test.cpp:122]: m_proxIntersection is created/instanciated in the Setup() function (for sure it wont be null) so I dont really see the point to test if it is null 🤔

File: modules/SofaMiscCollision/SofaMiscCollision_test/CMakeLists.txt (hunk 4 above)
Comment (fredroy) [reply modules/SofaMiscCollision/SofaMiscCollision_test/CMakeLists.txt:24]: yes, it allows the use of SofaMeshCollision_test includes into SofaMiscCollision (as there is no created cmake target for tests)

**Pull request merged**